File: apps/zLinebot/services/market-crawler/src/core/queue.py
import redis
import os
import json

r = redis.Redis.from_url(
    os.environ.get("REDIS_URL","redis://localhost:6379")
)

def enqueue(job):
    r.lpush("crawl_jobs", json.dumps(job))

def dequeue():
    job = r.rpop("crawl_jobs")
    if not job:
        return None
    return json.loads(job)

File: apps/zLinebot/services/tiktok-farm/src/core/queue.js
import { createClient } from "redis"

const client = createClient({
url: process.env.REDIS_URL
})

await client.connect()

export async function enqueue(job){

await client.lPush(
"video_jobs",
JSON.stringify(job)
)

}

export async function dequeue(){

const r = await client.rPop("video_jobs")

if(!r) return null

return JSON.parse(r)

}
